### PR TITLE
Fixes serious bug with the symbols parser (plus some docs)

### PR DESCRIPTION
**Pull request by tmbb**
I'm sorry this PR comes together with the docs for the Context module, but my Git-fu is not enough to change history and turn these two commits into two branches or something like that.

### CODE REVIEW

File: lib/ex_spirit/parser/text.ex [hunk 1]
```diff
@@ -530,13 +531,13 @@ defmodule ExSpirit.Parser.Text do
                 }
               parser when is_function(parser, 1) ->
                 %{context |
-                  position: context.position + byte_size(<<c::utf8>>),
+                  position: context.position,
```
Comment (OvermindDL1) [inline lib/ex_spirit/parser/text.ex]: As this line is a no-op now other than just causing an extra set (needless cost), can you just delete it entirely?  :-)

File: lib/ex_spirit/parser/text.ex [hunk 2]
```diff
@@ -530,13 +531,13 @@ defmodule ExSpirit.Parser.Text do
...
                   column: if(c===?\n, do: 1, else: context.column+1),
                   line: context.line + if(c===?\n, do: 1, else: 0),
                 } |> parser.()
               value ->
                 %{context |
-                  position: context.position + byte_size(<<c::utf8>>),
+                  position: context.position,
```
Comment (OvermindDL1) [inline lib/ex_spirit/parser/text.ex]: Delete this one too?  ^.^

File: lib/ex_spirit/parser/text.ex [hunk 3]
```diff
@@ -530,13 +531,13 @@ defmodule ExSpirit.Parser.Text do
                 }
               parser when is_function(parser, 1) ->
                 %{context |
-                  position: context.position + byte_size(<<c::utf8>>),
+                  position: context.position,
                   column: if(c===?\n, do: 1, else: context.column+1),
                   line: context.line + if(c===?\n, do: 1, else: 0),
```
Comment (OvermindDL1) [inline lib/ex_spirit/parser/text.ex]: Also wtf was I thinking here, can you delete `column` and `line` lines here too, these cases are only for when `c` is used, and as we are in the initial `nil` branch, then it is obviously not used, so these should be no-ops too!  o.O

File: lib/ex_spirit/parser/text.ex [hunk 4]
```diff
@@ -530,13 +531,13 @@ defmodule ExSpirit.Parser.Text do
...
                 } |> parser.()
```
Comment (OvermindDL1) [inline lib/ex_spirit/parser/text.ex]: So yeah, probably just turn this whole set into just `parser.(context)` or `context |> parser.()` or whatever.  >.>

File: lib/ex_spirit/parser/text.ex [hunk 5]
```diff
@@ -530,13 +531,13 @@ defmodule ExSpirit.Parser.Text do
...
               value ->
                 %{context |
-                  position: context.position + byte_size(<<c::utf8>>),
+                  position: context.position,
                   column: if(c===?\n, do: 1, else: context.column+1),
                   line: context.line + if(c===?\n, do: 1, else: 0),
```
Comment (OvermindDL1) [inline lib/ex_spirit/parser/text.ex]: More of wtf was I thinking!?!  o.O

Only thing this whole set should be doing is just setting the `result` and that is it.  ^.^;

**Review by OvermindDL1 — CHANGES_REQUESTED** [review]
I have some more WTF's on-my-part here too!  Can you fix them as well since they are all related with what you currently changed?  ^.^

_/me has no clue how they missed that `nil` branch not branching right and yet *still* using `c`, wtf..._

**Comment by OvermindDL1** [comment]
> I'm sorry this PR comes together with the docs for the Context module, but my Git-fu is not enough to change history and turn these two commits into two branches or something like that.

Nah all good, the commits are distinct and that is the important part.  :-)

But yeah, this whole `nil` branch is a large set of wtf's on my part, thanks for the catch!!!

**Comment by tmbb** [comment]
> But yeah, this whole nil branch is a large set of wtf's on my part, thanks for the catch!!!

I thought so... I was just trying to touch as few lines as possible :p

**Comment by tmbb** [comment]
Fixed.

**Comment by tmbb** [comment]
You should bump the version number, release a new version and anounce it in the forum topic. The bug is quite serious :)

**Comment by OvermindDL1** [comment]
Woot!  Looks great!

> You should bump the version number, release a new version and anounce it in the forum topic. The bug is quite serious :)

Yep yep, doing so now.  ^.^

Thanks much!!!!

**Pull request merged**